.vscode/settings.json

```diff
@@ -0,0 +1,5 @@
+{
```

Comment:
i wonder if IDE settings should be added to the repo. i have quite a big settings file, some of the stuff is just personal preference how the IDE behaves. isn't `.prettierrc.json` enough to make it work?

Reply:
I put the "non-negotiable" stuff in vscode, primarily, the tab sizing

And, its important that the autoformatting is turned on, otherwise prettier settings don't get applied

goal is to keep the vscode files minimal

Reply:
Also, this should preserve most of your existing settings 👍

Reply:
thing is, i have quite a lot settings in there which i would need to delete and/or migrate to my global settings (otherwise they would be enforced on every project member). that's why i wondered if the options in `settings.json` are needed to make prettier work in vscode or if putting the settings in `.prettierrc.json` is enough. Of course this wouldn't cover `formatOnSave`, however i would consider those kind of settings (how the editor behaves) personal preference (personally you can set it in the project or user settings anytime). In the end you only really enforce a style guide by failing it in CI or?

Reply:
If prettier doesn't work without the settings in the vscode config i'm fine with putting it on the repo. but then i'd suggest that we can / should get rid of `.prettierrc.json` as its just redundant settings.

Reply:
anyway, i looked at it again and i see that it make sense to have the editor tabSize set by default. i guess i can settle on migrating some of my settings to user config (since you can scope settings by language).

I'd suggest adding:
```
    "editor.insertSpaces": true,
    "editor.detectIndentation": false,
```

and possibly 

```
    "[typescript]": {
        "editor.rulers": [100]
   }
```

its also an option to put the prettier config in the vsocde settings:

```
    "prettier.tabWidth": 4,
    "prettier.singleQuote": true,
    "prettier.printWidth": 100,
```

Reply:
Actually, yeah the editor config is potentially too invasive, and I should really just add a lint stage to enforce. I'll remove the vscode files 👍